### PR TITLE
Add confluent-kafka-python to omnibus

### PR DESCRIPTION
**Pull request by yzhan289**
### What does this PR do?
This PR adds the `confluent-kafka-python` library to the omnibus, since the regular existing wheels on PyPI [do not include Kerberos support](https://github.com/confluentinc/confluent-kafka-python/tree/master#install)

### Motivation
Moving the current `kafka_consumer` check from `kafka-python` to `confluent-kafka-python`: https://github.com/DataDog/integrations-core/pull/13918

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: omnibus/config/software/librdkafka.rb
Comment (yzhan289) [inline omnibus/config/software/librdkafka.rb:15]: I wasn't too sure about how these commands should be, I just used the same commands in the `librdkafka` instructions: https://github.com/confluentinc/librdkafka#building

File: omnibus/config/software/datadog-agent-integrations-py2.rb
Comment (FlorentClarret) [inline omnibus/config/software/datadog-agent-integrations-py2.rb:65]: I don't think we should modify this file since we're not going to support py2

File: omnibus/config/software/librdkafka.rb
Comment (FlorentClarret) [reply omnibus/config/software/librdkafka.rb:16]: From what I understand [here](https://github.com/confluentinc/confluent-kafka-python/blob/master/INSTALL.md#install-from-source), we do not have to build `librdkafka` from the source, only `confluent-kafka` needs to be built from the source, we can install it using apt/brew/yum/vcpkg following this docs: https://github.com/confluentinc/confluent-kafka-python/blob/master/INSTALL.md#install-from-source

File: omnibus/config/software/librdkafka.rb
Comment (bkabrda) [inline omnibus/config/software/librdkafka.rb:1]: Dependencies that don't go in the Agent packages should be added in https://github.com/DataDog/datadog-agent-buildimages/ (and for macOS to https://github.com/DataDog/datadog-agent-macos-build/blob/master/scripts/builder_setup.sh).

Just a note here that I'll leave for your consideration: It seems that you define a `default_version`, its `sha256` etc, but then proceed to install via vcpkg/yum/... This basically means that you get a random version that doesn't necessarily correspond to `default_version`. This might or might not be ok depending on librdkafka compatibility guarantees and I'll leave that up to you - I want to highlight this so that we know it's an intentional and explicit choice.

Comment (FlorentClarret) [reply omnibus/config/software/librdkafka.rb:1]: I'm gonna have a deper look at their doc and try some stuff locally and I'll let you know. Thanks again for your help!

**Review by alai97 — APPROVED** [review]
Looks good for docs!

File: .gitlab-ci.yml
Comment (amenasria) [inline .gitlab-ci.yml:142]:
```suggestion
  DATADOG_AGENT_BUILDIMAGES: v13660599-7e22554
```

File: .gitlab-ci.yml
Comment (amenasria) [inline .gitlab-ci.yml:145]:
```suggestion
  DATADOG_AGENT_ARMBUILDIMAGES: v13660599-7e22554
```

**Review by amenasria — COMMENTED** [review]
We'd prefer having the detailed image version, otherwise LGTM (once the runtime dependencies problem is settled)

File: .gitlab-ci.yml
Comment (FlorentClarret) [inline .gitlab-ci.yml:145]: I'll update the tags once I'll be done with the buildimages repo 🙂

File: release.json
Comment (ogaca-dd) [inline release.json:21]: Is this change expected?

**Review by ogaca-dd — DISMISSED** [review]

**Review by cedricvanrompay-datadog — APPROVED** [review]

File: release.json
Comment (FlorentClarret) [reply release.json:21]: Marking this PR to draft since this is not ready yet. We modified this to be able to use [this updated recipe](https://github.com/DataDog/omnibus-software/pull/439) that is not merged yet to build the docker image in this CI for testing purpose

File: omnibus/config/software/confluent-kafka-python.rb
Comment (FlorentClarret) [inline omnibus/config/software/confluent-kafka-python.rb:24]: The OOTB confluent-kafka wheels already provides librdkafka (but whitout kerberos support ofc) so we can simply `pip install` it and get rid of everything else. You can also move the `dependency "librdkafka"` to the else and drop the windows part from the `librdkafka.rb` file

Comment (yzhan289) [reply omnibus/config/software/confluent-kafka-python.rb:24]: Ah yeah I added this back in my commit changing the `pip` command to see if the `INCLUDE` or `LIB` flags were being used in some way since the CI was originally failing. I'll apply your suggestions now since the CI is passing!

Comment (yzhan289) [reply omnibus/config/software/confluent-kafka-python.rb]: Applied: https://github.com/DataDog/datadog-agent/pull/15470/commits/355b11a31a93832759c94de8e3d626c94c23edef

File: omnibus/config/software/lmdb.rb
Comment (KSerrania) [inline omnibus/config/software/lmdb.rb:8]: I think this should be:
```suggestion
relative_path "lmdb-LMDB_#{version}/libraries/liblmdb"
```

since expanding the `LMDB_0.9.29.tar.gz` archive puts the contents in a directory named `lmdb-LMDB_0.9.29`.

Comment (yzhan289) [reply omnibus/config/software/lmdb.rb:8]: Ah gotcha, thanks! I'll try this.

File: release.json
Comment (yzhan289) [reply release.json]: Fixed! We will not need gdbm anymore.

**Review by bkabrda — CHANGES_REQUESTED** [review]
The PR looks really good now, but we're missing license and license_file for one of the recipes. I can approve once that is added.

**Review by bkabrda — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**
